### DISCUSSION TITLE
Cannot push to remote

### DISCUSSION THREAD

**Discussion by ludojudo** [community/community#40163] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Dear Community,

I am a new user and am encountering issues trying to push some code to remote:

I am following the below tutorial on how to interact with git on windows using git bash:

https://www.simplilearn.com/tutorials/git-tutorial/git-installation-on-windows

everything works well until step 12, when I try to execute the command: `push origin master` I get the message "... does not appear to be a git repository, make sure you have the correct access right and that the repository exists"

I decided to try downloading github desktop to see if that would work. This included logging in on the app and all seems authenticated. However when I try to "publish branch to remote" I also get an error message (see below).

It looks like I am getting something really basic wrong, but can't see what it is. Any help would be much appreciated.

PS it looks like the the commits I made on bash are visible to github desktop, so it is just a case of uploading to remote.

![image](https://user-images.githubusercontent.com/20744247/204097747-a2c1ca63-e446-4e3f-91ac-d838df5d84a4.png)

![image](https://user-images.githubusercontent.com/20744247/204097762-43697fb2-40cc-424f-9a66-f990024ee54e.png)

**Comment by masayukipy** [comment]
You have to make sshkey as following.
https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent
And then add to publish key to github.
Open git bash.
command "eval $(ssh-agent)"
Then command "ssh-add /your privatekey location"    e.g. /f/sshkey/newkey
Input password
Then push

**Reply by ludojudo** [reply]
Hi @maiermarc75 

Thanks a lot for your reply!

I had a further look on stackoverflow where someone recommended [https://stackoverflow.com/questions/25927914/git-error-please-make-sure-you-have-the-correct-access-rights-and-the-reposito](url) running the following command to solve a similar issue to mine:

`git remote set-url origin https://github.com/username/repository.git`

After executing this command when trying to push I got a popup asking me to authenticate via browser, and I am now able to push things to remote from bash, and my github desktop is also not complaining.

I think that I have managed to fix the `https` method of communicating with remote, while your solution relies on the alternative `ssh` protocol. Hopefully I have done enough to use github now.

**Reply by masayukipy** [reply]
Thanks for your kindly.
